IpmiFeaturePkg/IpmiPowerRestorePolicy/UnitTest/TestIpmiPowerRestorePolicyHost.c

```diff
@@ -0,0 +1,380 @@
+/** @file TestIpmiPowerRestorePolicyHost.c
+
+  Copyright (c) Microsoft Corporation.
+  PDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <stdio.h>
+#include <string.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/UnitTestLib.h>
+
+//
+// Common header files
+//
+#include <PiPei.h>
+#include <Library/DebugLib.h>
+#include <Library/IpmiCommandLib.h>
+#include <Library/PlatformPowerRestorePolicyConfigurationLib.h>
+
+#define UNIT_TEST_NAME     "TestIpmiPowerRestorePolicyHost"
+#define UNIT_TEST_VERSION  "1.0"
+
+//
+// Power Restore Policy configuration.
+//
+#define  POWER_RESTORE_POLICY_POWER_OFF   0x00

```

Comment:
nit:

Redefinition of definese from PlatformPowerRestorePolicyConfigurationLib.h